[Flaky] rspec ./spec/services/tax_rate_finder_spec

### What we should change and why (this is tech debt)
<!-- The use of X changed and part y is now redundant. We can remove it and... -->
Failed examples:

rspec ./spec/services/tax_rate_finder_spec.rb:25 # TaxRateFinder getting the corresponding tax rate finds the tax rate of a shipping fee

```
Stopped processing SimpleCov as a previous error not related to SimpleCov has been detected
Coverage report generated for RSpec to /home/runner/work/openfoodnetwork/openfoodnetwork/coverage. 7625 / 25321 LOC (30.11%) covered.
/opt/hostedtoolcache/Ruby/3.1.4/x64/bin/ruby -I/home/runner/work/openfoodnetwork/openfoodnetwork/vendor/bundle/ruby/3.1.0/gems/rspec-core-3.12.2/lib:/home/runner/work/openfoodnetwork/openfoodnetwork/vendor/bundle/ruby/3.1.0/gems/rspec-support-3.12.1/lib /home/runner/work/openfoodnetwork/openfoodnetwork/vendor/bundle/ruby/3.1.0/gems/rspec-core-3.12.2/exe/rspec  --format progress --format KnapsackPro::Formatters::TimeTracker --default-path spec "spec/services/order_invoice_comparator_spec.rb" "spec/requests/api/orders_spec.rb" "spec/jobs/subscription_placement_job_spec.rb" "spec/services/process_payment_intent_spec.rb" "spec/services/place_proxy_order_spec.rb" "spec/services/order_available_shipping_methods_spec.rb" "spec/views/spree/shared/_order_details.html.haml_spec.rb" "spec/services/order_cycle_distributed_products_spec.rb" "spec/services/tax_rate_finder_spec.rb" "spec/services/shop/order_cycles_list_spec.rb" "spec/services/order_cart_reset_spec.rb" "spec/requests/api/routes_spec.rb" "spec/services/order_payment_finder_spec.rb" "spec/services/order_cycle_webhook_service_spec.rb" "spec/helpers/bulk_form_builder_spec.rb" "spec/services/order_data_masker_spec.rb" "spec/helpers/shop_helper_spec.rb" "spec/queries/payments_requiring_action_spec.rb" "spec/services/image_importer_spec.rb" "spec/services/customer_order_cancellation_spec.rb" "spec/helpers/admin/enterprises_helper_spec.rb" "spec/mailers/test_mailer_spec.rb" "spec/reflexes/user_reflex_spec.rb" "spec/views/admin/products_v3/_filters.html.haml_spec.rb" "spec/jobs/webhook_delivery_job_spec.rb" "spec/services/path_checker_spec.rb" "spec/jobs/job_logger_spec.rb" "spec/helpers/spree/admin/general_settings_helper_spec.rb" "spec/services/product_filters_spec.rb" "spec/requests/checkout/stripe_sca_spec.rb" failed
Error: Process completed with exit code 1.
```


### Context
<!-- This came up in PR #3451#issuecomment-461616636. -->
https://github.com/openfoodfoundation/openfoodnetwork/actions/runs/7569000531/job/20611382296


### Impact and timeline
<!-- Example: This simplifies the upgrade to Spree 3 and should be done after upgrading to Spree 2. -->

--- a/spec/factories/tax_category_factory.rb
+++ b/spec/factories/tax_category_factory.rb
@@ -2,7 +2,7 @@
 
 FactoryBot.define do
   factory :tax_category, class: Spree::TaxCategory do
-    name { "TaxCategory - #{rand(999_999)}" }
+    sequence(:name) { |n| "TaxCategory - #{n}" }
     description { generate(:random_string) }
   end
 end

--- a/spec/factories/zone_factory.rb
+++ b/spec/factories/zone_factory.rb
@@ -2,7 +2,7 @@
 
 FactoryBot.define do
   factory :zone, class: Spree::Zone do
-    name { generate(:random_string) }
+    sequence(:name) { |n| "#{generate(:random_string)}#{n}" }
     description { generate(:random_string) }
   end
 

